### ISSUE TITLE
[Java] False negative (LawOfDemeter): Try-resource block treated as function call by rule. 

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2192]
**Affects PMD Version:** 
6.20.0.
**Rule:**
LawOfDemeter

**Description:**
False positive. Object created elsewhere is not flagged within try-resource block but is flagged outside of it.

**Code Sample demonstrating the issue:**

```
package pmdtests;

import java.io.IOException;
import java.io.PrintWriter;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

// @WebServlet("/DbServlet")
public class DbServlet extends HttpServlet {
  private static final long serialVersionUID = 1L;
  private static final String URL = "jdbc:derby:HomeDB";
  private transient Connection link;

  public void doPost(HttpServletRequest request, HttpServletResponse response)
      throws ServletException, IOException {
    response.setContentType("text/HTML");
    PrintWriter out = response.getWriter();
    
    try (Statement statement = link.createStatement();
         ResultSet results = statement.executeQuery("SELECT * FROM PhoneNums");) {
      
      while (results.next()) {
        out.println("<TR>");
        out.println("<TD>");
        out.println(results.getString("Surname"));
        out.println("</TD>");
        out.println("<TD>");
        out.println(results.getString("Forenames"));
        out.println("</TD>");
        out.println("<TD>");
        out.println(results.getString("PhoneNum"));
        out.println("</TD>");
        out.println("</TR>");
      }
      out.println("</TABLE>");
    } catch (SQLException sqlEx) {
      printHtmlSelectError(out);
      return;
    }
    out.println("<BODY>");
    out.println("</HTML>");
    out.flush();
  }
}

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0

Similarly to #3801, the code would not flag under the current implementation, regardless of try-with-resources. `Statement statement = link.createStatement()` is at distance 1, making `ResultSet results = statement.executeQuery("SELECT * FROM PhoneNums")` be at distance 2… but all calls on `results` are pure data (retrieves primitives, Strings or boxed types) which the implementation allows for.

Even if the code is modified to introduce a new indirection level, the use of try-with-resources is irrelevant to the current rule implementation.
